src/transsystem.c

```diff
@@ -45,18 +45,22 @@ callback_tree_before(T_PTR_tree entry) {
 
   info = (char*) tree_getinfo(entry);
   if (strcmp(info,"rules") == 0) {
+  	printf("Entering rules section\n");
     tree_brk_branch(entry);
     /* nbr_rules will be fill at this point */
-    rulescode_produce(entry, &_system);
+    rulescode_produce(entry, _system);
```

Comment:
We needed `&` before this change but I don't see clearly why now `_system` and not `&_system` is enough?
Is it because we already allocated the memory?
Also `&_init` and `&_unsafe` have stayed, can't `&` be also removed following the same principle? 
Is it because we haven't yet allocated the memory?
If  so then perhaps it is better to do allocation in one place only and have `&`less arguments everywhere.